R/rmarkdown.R

```diff
@@ -61,6 +61,21 @@ render_rmarkdown <- function(pkg, input, output, ..., copy_images = TRUE, quiet
   if (copy_images) {
     ext <- rmarkdown::find_external_resources(input_path)
 
+    # temporarily copy the rendered html into the input path directory and scan
+    # again for additional external resources that may be been referenced by R code
+    ext2 <- withr::with_tempfile(
```

Comment:
This feels a bit of an abuse of `with_tempfile()`. I think it would be cleaner to just `file.copy()` in and then `with::defer(unlink())`.  

Is it necessary to scan both the input and output files?

Reply:
> This feels a bit of an abuse of `with_tempfile()`.

I agree, I rewrote it without using `with_tempfile()`, creating the file with `file_temp()` instead. I'm using the random filename just to be certain we don't stomp on anything created be the user, e.g. html files with the same base name as the article input.

> Is it necessary to scan both the input and output files?

Currently pkgdown misses resources that are added in R chunks or by R code. The advice given is to use the `resource_files` YAML header, but it can be hard to notice these files are missing in the first place. I've been bitten by this before. 

By doing another scan of the rendered HTML file in the source folder context, we can pick up additional low-hanging fruit resources that might exist in the rendered version but aren't statically identifiable from the source.

Btw, I committed the test separately in 4b20b8015ffe07db4a6eddd5bb199c726b8d3b2a so that you could checkout that commit and see an example where the current process misses an external resource.

Reply:
Sorry, I should've been clearer: can we just scan the output file? What will that miss?

Reply:
ah that makes sense! Unfortunately, I think we'll end up missing the resources explicitly declared in the YAML if we only look at the output.